base/fs.jl

```diff
@@ -163,6 +164,15 @@ end
 @windowsxp_only symlink(p::AbstractString, np::AbstractString) =
     error("WindowsXP does not support soft symlinks")
 
+@non_windowsxp_only function readlink(path::AbstractString)
+    l = ccall(:jl_fs_readlink, Ptr{UInt8}, (Ptr{UInt8}, ), path)
+    linkval = bytestring(l)
+    Libc.free(l)
+    return linkval
+end
+@windowsxp_only readlink(p::AbstractString, np::AbstractString) =
+error("WindowsXP does not support soft symlinks")
```

Comment:
Unless the underlying libuv function errors on XP, there's no reason for us to make `readlink` error there.